activerecord/lib/arel/visitors/to_sql.rb

```diff
@@ -21,11 +21,15 @@ def compile(node, collector = Arel::Collectors::SQLString.new)
       private
         def visit_Arel_Nodes_DeleteStatement(o, collector)
           o = prepare_delete_statement(o)
-
+          has_table_alias(o)
```

Comment:
I added a puts here to make sure the table alias is working and I don't think it is getting through to Arel. For sqlite and mysql it was `nil`. It looks like active record isn't passing this along to the relation because klass is always nil in Arel::Table.

Reply:
I double checked this tonight and now it outputs the `table_alias` with the updated method noted above.